### PR TITLE
Inline template in map-control directive - Fix Issue #1590

### PR DESCRIPTION
**Pull request by mehanika**
You will be able to use an inline template as content for the map control. Example:

```html
<ui-gmap-map-control  position="top-right" index="1">
          <div ng-controller="mapWidgetCtrl">
            <button class="btn btn-lg btn-primary"  ng-click="clearWidget.controlClick()">{{clearWidget.controlText}}</button>
          </div>
</ui-gmap-map-control>
```
And also you can still use the attribute `template`. This PR closes the issue #1590 

**The only thing missing is a PR for updating the documentation**

### CODE REVIEW

**Comment by cthrax** [comment]
My memory isn't as good as @nmccready but I believe transclusion used to be a feature and it was removed because it became unmanageable to synchronize the transcluded content and the marker DOM. Since the google maps API is not aware or part of the angular $digest cycle. I may be misremembering though.

**Comment by nmccready** [comment]
@cthrax yeah that's correct for plural directives, but this is something that I don't think will be large for ng-repeat.

**Comment by cthrax** [comment]
Alright, @mehanika, if you want to give us a PR with all the build artifacts, and a documentation update, we've got ourselves a new feature. Thanks a lot for the contribution!

**Comment by mehanika** [comment]
@cthrax I'm glad to contribute  :). For documentation update, to which branch should I make the PR?

**Comment by nmccready** [comment]
@mehanika `website` branch.

**Comment by mehanika** [comment]
I'm having an issue running `bower instal`in `website` branch. My console responds `ENORESTARGET  No tag found that was able to satisfy ~0.3.0`, it seems to be `share-button` dependency

**Comment by cthrax** [comment]
Hmmm, is it possibly your bower install is out of date? https://github.com/bower/bower/pull/669

File: dist/.htaccess [hunk 1]
```diff
@@ -0,0 +1,543 @@
+# Apache Configuration File
```
Comment (nmccready) [inline dist/.htaccess]: What change added this file? This seems like an accident.

Comment (mehanika) [inline dist/.htaccess]: I think was an accident, I will remove it

**Comment by mehanika** [comment]
@cthrax @nmccready I found a bug in the directive while I was updating the documentation :(

**Comment by cthrax** [comment]
:cry:

**Comment by mehanika** [comment]
@cthrax I have already solve the bug. Also I'm still having problems with `share-button` dependency (in website branch). I ran `bower info share-button` and gave this 
```
Available versions:
  - 1.0.3
```
This breaks `share` directive

**Comment by cthrax** [comment]
Wow, that hasn't been touched in awhile. Apparently we started using it around 0.0.3? If you want to create a branch from the website "mehanika/share-button" or something, I can help you take a look at that.

First though, you might try setting the version to =0.0.3. According to their github repo, that is still available.

**Comment by mehanika** [comment]
@cthrax  I've already created a branch named [share-button](https://github.com/mehanika/angular-google-maps/tree/share-button). I've setted the version to =0.0.3 but I get the same error.

**Comment by cthrax** [comment]
@mehanika Why can't these things just go smooth? :) I was able to get passed that resolution by making the dependency 

`"share-button": "git+https://github.com/carrot/share-button.git#v0.0.3" `

Hope that fixes the issue for you. :-\

**Comment by mehanika** [comment]
Yes, that fixed the issue, thanks. Do I need to update  angular-google-maps version  in `share-button` branch? If not, what's the next step take? Only create the PR for updating the docs?

**Comment by cthrax** [comment]
@mehanika the website points to rawgit for it's version, so once we get things merged, it should be using the latest. So next steps are looking at getting this merged, @nmccready may need your help there. Do we merge from master onto 2.1.x? Or should this be PR'd against 2.1.x?

**Comment by mehanika** [comment]
@cthrax I made a PR for the documentation #1815

**Comment by marysgonzalez** [comment]
It doesn't work for me, I still getting:

mapControl: could not find a valid template property

**Comment by mehanika** [comment]
@marysgonzalez Could you create a jsbin to check ?. Also you can see it in action in the examples, run `grunt example` and enter to [http://localhost:3100/example/dynamic-map-height.html](http://localhost:3100/example/dynamic-map-height.html).

**Comment by marysgonzalez** [comment]
<ui-gmap-google-map id="mapa" center="map.center" zoom="map.zoom">
          <ui-gmap-map-control  position="top-right" index="1">
          <div ng-controller="MapaController">
             <button class="btn btn-circle btn-primary" ng-click="addReport();"><span style="font-weight: 700">+</span></button>
          </div>
        </ui-gmap-map-control>
	</ui-gmap-google-map> 

Err: mapControl: could not find a valid template property.

I am sorry, this is all I can provide for now

**Comment by mehanika** [comment]
@marysgonzalez I think you are not using the branch `extended-map-control` beacuse the error you are getting is old. As you can see in [control.coffe](https://github.com/mehanika/angular-google-maps/blob/extended-map-control/src/coffee/directives/api/control.coffee#L17) the error you should be getting is:

`mapControl: could not find a valid template property or elements for transclusion`

**Comment by nmccready** [comment]
@cthrax is there any reason we should not merge this? I have not been following this closely. Only main hold up I see now is  it needs to be updated with latest code.

**Comment by cthrax** [comment]
@nmccready Once the conflicts are resolved, I think it's good to go.

**Comment by nmccready** [comment]
@mehanika ping

**Comment by mehanika** [comment]
@nmccready It's already up to date

**Comment by cthrax** [comment]
Merged, thanks @mehanika !

**Comment by mehanika** [comment]
I'd like to make an update to this, I have added an usless parent div to the map-control directive (when you use inline template) and I've removed it [view commit](https://github.com/mehanika/angular-google-maps/commit/378c5533cda5aeb1f6d65aec59eb144c3094a747)

**Pull request merged**